From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alistair Delva <adelva@google.com>
Date: Mon, 29 Mar 2021 11:50:21 -0700
Subject: ANDROID: GKI: Enable POWER_RESET_SYSCON

Enable support for board reset with "generic" syscon register mappings.

This feature is needed by CONFIG_SYSCON_REBOOT_MODE which some boards
may build as a module to communicate the reboot mode back to the
bootloader.

This feature is used by rockchip platforms such as Rock Pi 4.

Bug: 183962924
Change-Id: Icfdebf29777533799e238fdadf1de064f271c856
Signed-off-by: Alistair Delva <adelva@google.com>
---
 arch/arm64/configs/gki_defconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -365,6 +365,7 @@ CONFIG_SPMI=y
 # CONFIG_PINCTRL_SUN50I_H6_R is not set
 CONFIG_GPIO_GENERIC_PLATFORM=y
 CONFIG_POWER_RESET_HISI=y
+CONFIG_POWER_RESET_SYSCON=y
 # CONFIG_HWMON is not set
 CONFIG_THERMAL=y
 CONFIG_THERMAL_NETLINK=y
@@ -379,7 +380,6 @@ CONFIG_THERMAL_EMULATION=y
 CONFIG_WATCHDOG=y
 CONFIG_WATCHDOG_CORE=y
 CONFIG_MFD_ACT8945A=y
-CONFIG_MFD_SYSCON=y
 CONFIG_REGULATOR=y
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
 CONFIG_RC_CORE=y
